tests/30rooms/05aliases.pl

```diff
@@ -226,6 +226,37 @@ sub _test_can_create_and_delete_alias {
       })
    };
 
+test "Users with sufficient power-level can delete other's aliases",
+   requires => [ $creator_fixture, $room_fixture, local_user_fixture(), room_alias_fixture(),
+                 qw( can_create_room_alias )],
+
+   do => sub {
+      my ( $user, $room_id, $other_user, $room_alias ) = @_;
+      my $server_name = $user->http->server_name;
+
+      matrix_change_room_power_levels(
+         $user, $room_id, sub {
+            $_[0]->{users}->{$other_user->user_id} = 100;
+         },
+      )->then( sub {
+         do_request_json_for( $user,
+           method => "PUT",
+           uri    => "/r0/directory/room/$room_alias",
+
+           content => { room_id => $room_id },
+         )
+      })->then( sub {
+         do_request_json_for( $other_user,
+           method => "DELETE",
+           uri    => "/r0/directory/room/$room_alias",
+
+           content => {},
+         )
+      })->then( sub {
+         Future->done(1);
```

Comment:
this is a no-op

Reply:
maybe we should request the alias with `$user` and check that it has been deleted?

Reply:
Having a final `Future->done(1);` seems to be common in this file. Should they be removed?

For this particular test we can assert something here, yeah.

Reply:
If I remove this then I get an error and the test fails:

```
Expected __ANON__(tests/30rooms/05aliases.pl line 257) to return a Future at /sytest/run-tests.pl line 745.
```

Reply:
I figured out a better way to handle this.